utils/retry/retryexecutor.go

```diff
@@ -0,0 +1,58 @@
+package retry
+
+import (
+	"fmt"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"time"
+)
+
+const retryMessage = "%sAttempt %v - %s"
+
+type ExecutionHandlerFunc func() (bool, error)
+
+type RetryExecutor struct {
+	// The amount of retries to perform.
+	MaxRetries int
+
+	// Time period between retries.
+	RetriesInterval int
+
+	// Message to display when retrying.
+	RetryMessage string
+
+	// Prefix to print at the beginning of each log.
+	LogMsgPrefix string
+
+	// ExecutionHandler is the operation to run with retries.
+	ExecutionHandler ExecutionHandlerFunc
+}
+
+func (runner *RetryExecutor) Execute() error {
+	var err error
+	var shouldRetry bool
+	for i := 0; i <= runner.MaxRetries; i++ {
+		// Run ExecutionHandler
+		shouldRetry, err = runner.ExecutionHandler()
+
+		// If should not retry, return
+		if !shouldRetry {
+			return err
+		}
+
+		log.Warn(runner.getLogRetryMessage(i, err))
+		// Going to sleep for RetryInterval seconds
+		if i < runner.MaxRetries {
```

Comment:
To make sure that we do not call the sleep function with a zero interval value:
if i < runner.MaxRetries { --> if runner.RetriesInterval) > 0 && i < runner.MaxRetries {

Reply:
Fixed.